adapters/thetradedesk/thetradedesk.go

```diff
@@ -0,0 +1,190 @@
+package thetradedesk
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"text/template"
+
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+)
+
+const PREBID_INTEGRATION_TYPE = "1"
+
+type adapter struct {
+	bidderEndpoint string
+}
+
+type ExtImpBidderTheTradeDesk struct {
+	adapters.ExtImpBidder
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	pubID, err := getPublisherId(request.Imp)
+
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	modifiedImps := make([]openrtb2.Imp, 0, len(request.Imp))
+
+	for _, imp := range request.Imp {
+
+		if imp.Banner != nil {
+			if len(imp.Banner.Format) > 0 {
+				firstFormat := imp.Banner.Format[0]
+				bannerCopy := *imp.Banner
+				bannerCopy.H = &firstFormat.H
+				bannerCopy.W = &firstFormat.W
+				imp.Banner = &bannerCopy
+
+			}
+		}
+
+		modifiedImps = append(modifiedImps, imp)
+	}
+
+	request.Imp = modifiedImps
+
+	if request.Site != nil {
+		siteCopy := *request.Site
+		if siteCopy.Publisher != nil {
+			publisherCopy := *siteCopy.Publisher
+			publisherCopy.ID = pubID
+			siteCopy.Publisher = &publisherCopy
```

Comment:
should be test on it.
![image](https://github.com/user-attachments/assets/7dacdbc3-e516-414a-914c-4d8b8f442cdb)